beetsplug/unimported.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function
+import os
+
+from beets import util
+from beets.plugins import BeetsPlugin
+from beets.ui import Subcommand
+
+
+class Unimported(BeetsPlugin):
+
+    def __init__(self):
+        super(Unimported, self).__init__()
+        self.config.add(
+            {
+                'ignore_extensions': []
+            }
+        )
+
+    def commands(self):
+        def print_unimported(lib, opts, args):
+            exts_to_ignore = self.config['ignore_extensions'].as_str_seq()
+            in_folder = set(
+                (os.path.join(r, file) for r, d, f in os.walk(lib.directory)
+                 for file in f if not any(
+                    [file.endswith(extension.encode()) for extension in
```

Comment:
It might be slightly better to do all the `.encode()`ing up front, so it doesn't have to happen repeatedly for every file?

Also, it's perhaps worth noting that this will match `foojpg`, for example—not just `foo.jpg`.